package NKikimrBlobStorage;
option java_package = "ru.yandex.kikimr.proto";

// Defines parameters of the drive
message TDriveModel {
    enum EModelSource {
        SourceClear = 0;
        SourceGenericHardcode = 1;
        SourceMatchingIdHardcode = 2;
        SourceLocalMeasure = 3;
        SourceConfigForMatchingId = 4;
    };

    // Model data
    optional uint64 SeekTimeNsec = 10 [default = 0];
    // optional uint64 StepSizeBytes = 20 [default = 0]; // Not used anymore
    optional uint64 BulkReadBlockSizeBytes = 30 [default = 0];
    optional uint64 BulkWriteBlockSizeBytes = 40 [default = 0];
    optional uint64 TrimSpeedBps = 50 [default = 0];
    optional uint64 TotalChunksCount = 60 [default = 0];
    optional uint64 SpeedBpsRead = 70 [default = 0];
    optional uint64 SpeedBpsWrite = 80 [default = 0];
    optional uint64 SpeedBpsMinRead = 90 [default = 0];
    optional uint64 SpeedBpsMinWrite = 100 [default = 0];
    optional uint64 SpeedBpsMaxRead = 110 [default = 0];
    optional uint64 SpeedBpsMaxWrite = 120 [default = 0];
    optional uint64 OptimalQueueDepthRead = 130 [default = 0];
    optional uint64 OptimalQueueDepthWrite = 140 [default = 0];
    optional uint64 GlueingDeadlineRead = 150 [default = 0];
    optional uint64 GlueingDeadlineWrite = 160 [default = 0];

    // Model metadata
    optional EModelSource ModelSource = 1010 [default = SourceClear];
    optional string SourceModelNumber = 1020 [default = ""];
    optional string SourceFirmwareRevision = 1030 [default = ""];
    optional string SourceSerialNumber = 1040 [default = ""];
    optional bool IsSourceWriteCacheEnabled = 1050 [default = false];
    optional bool IsSourceSharedWithOs = 1060 [default = false];
    optional bool IsSourceSolidState = 1070 [default = false];
};

message TDriveModelList {
    repeated TDriveModel DriveModel = 10;
};

